File: DOTNETPrototype/Models/ServicePurchase.cs
namespace Models;

public partial class ServicePurchase
{
    public int Id { get; set; }

    public int Service { get; set; }

    public int User { get; set; }

    public DateTime Date { get; set; }

    public string? Status { get; set; }

    public int TemporalTokenCost { get; set; }

    public virtual Service ServiceNavigation { get; set; } = null!;

    public virtual User UserNavigation { get; set; } = null!;
}

File: MilestoneThreeWebsite/server/models/ServicePurchase.js
const mongoose = require('mongoose')

const ServicePurchaseSchema = new mongoose.Schema({
    service: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Service',
        required: true,
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    date: {
        type: Date,
        required: true
    },
    status: {
        type: String,
        required: true
    },
    temporalTokenCost: {
        type: Number,
        required: true
    }
})

const ServicePurchase = mongoose.model("ServicePurchase", ServicePurchaseSchema)
module.exports = ServicePurchase